karma/test-services.coffee

```diff
@@ -630,16 +633,294 @@ describe('services:', () ->
+    describe('get', () ->
+      it('gets a pod', () ->
+        expect(PodApiService.get.fn(21, 23)).toEqual({
+          method: 'GET',
+          url: "/pods/read/21/",
+          params: {case_id: 23}
+        })
+      )
+    )
+
+    describe('trigger', () ->
+      it('triggers an action', () ->
+        expect(PodApiService.trigger.fn(21, 23, 'foo', {bar: 'baz'})).toEqual({
+          method: 'POST',
+          url: "/pods/action/21/",
+          data: {
+            case_id: 23
+            action: {
+              type: 'foo',
+              payload: {bar: 'baz'}
+            }
+          }
+        })
+      )
+    )
+  )
+
+  #=======================================================================
+  # Tests for ModalService
+  #=======================================================================
+  describe('ModalService', () ->
+    ModalService = null
+
+    beforeEach(inject((_ModalService_) ->
+      ModalService = _ModalService_
+    ))
+
+    describe('confirm', () ->
+      describe('if no template url is given', () ->
+        it('should draw the modal', () ->
+          ModalService.confirm({
+            title: 'Foo',
+            prompt: 'Bar?'
+          })
+
+          $rootScope.$apply()
+
+          expect(document.querySelector('.modal-title').textContent)
+            .toMatch('Foo')
+
+          expect(document.querySelector('.modal-body').textContent)
+            .toMatch('Bar?')
+        )
+
+        it('should fulfill if the modal is accepted', () ->
+          fulfilled = false
+
+          ModalService.confirm({
+            title: 'Foo',
+            prompt: 'Bar?'
+          })
+          .then(-> fulfilled = true)
+
+          $rootScope.$apply()
+          expect(fulfilled).toBe(false)
+
+          angular.element(document.querySelector('.btn-modal-accept'))
+            .triggerHandler('click')
+
+          $rootScope.$apply()
+          expect(fulfilled).toBe(true)
+        )
+
+        it('should reject if the modal is cancelled', () ->
+          rejected = false
+
+          ModalService.confirm({
+            title: 'Foo',
+            prompt: 'Bar?'
+          })
+          .catch(-> rejected = true)
+
+          $rootScope.$apply()
+          expect(rejected).toBe(false)
+
+          angular.element(document.querySelector('.btn-modal-cancel'))
+            .triggerHandler('click')
+
+          $rootScope.$apply()
+          expect(rejected).toBe(true)
+        )
+      )
+
+      describe('if a template url is given', () ->
+        it('should draw the modal', () ->
+          ModalService.confirm({
+            templateUrl: '/sitestatic/templates/modals/dummy-confirm.html',
+            context: {title: 'Foo'}
+          })
+
+          $rootScope.$apply()
+
+          expect(document.querySelector('.modal-title').textContent)
+            .toMatch('Foo')
+
+          expect(document.querySelector('.modal-body').textContent)
+            .toMatch('Are you sure you want to do this?')
+        )
```

Comment:
@rowanseymour I ended up modifying the approach we discussed a bit for modals. Is it supporting both the prompt-strings approach and the template url approach. What ended up differently though, was that instead of providing a prompt template url, one can provide a template url for the entire modal. The reason for this is that it gives us more flexibility for controlling how the modal looks. Thought I should bring this up since it is a slight change in what we discussed.